nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/document/AddPermission.java

```diff
@@ -110,8 +114,8 @@ protected void addPermission(DocumentModel doc) {
         }
 
         String username = user;
-        if (username == null) {
-            username = NuxeoPrincipal.computeTransientUsername(email);
+        if (!isUserFound(username)) {
```

Comment:
Agree or not, the point is that we can share a document with someone not registered in Nuxeo, by using only an email :)

See https://doc.nuxeo.com/userdoc/managing-permissions/#granting-permissions-to-external-users-instant-share- for the functional use case.